responses/validate_response.go

```diff
@@ -38,7 +38,15 @@ func ValidateResponseSchema(
 	schema *base.Schema,
 	renderedSchema,
 	jsonSchema []byte,
+	cfg ...Option,
 ) (bool, []*errors.ValidationError) {
+
+	// Apply config options
+	config := configOptions{} // Default config
+	for _, opt := range cfg {
+		opt(&config)
+	}
```

Comment:
If you repeat this everywhere, I would like to suggest you to create a method

Something like

```go
func getConfig(opt ...Option) Config {
           // Apply config options 
         config := configOptions{} // Default config 
         for _, opt := range cfg { 
                 opt(&config) 
         }
}
```

Or this if you want avoiding the trailing ... each time you pass op

```go
func getConfig(opt []Option) Config {
           // Apply config options 
         config := configOptions{} // Default config 
         for _, opt := range cfg { 
                 opt(&config) 
         }
}
```

Reply:
Part of my drive here is to be backward compatible and to honor the current packaging and associated visibility.

So whilst I'd agree there is duplicative code most of these changes affect separate packages with their own (potentially unique) config  options so lifting it out probably requires a more holistic overhaul.

Reply:
ok, I get that